optuna/samplers/tpe/sampler.py

```diff
@@ -520,7 +521,10 @@ def _get_observation_pairs(study, param_name):
         elif trial.state is structs.TrialState.PRUNED:
             if len(trial.intermediate_values) > 0:
                 step, intermediate_value = max(trial.intermediate_values.items())
-                score = (-step, sign * intermediate_value)
+                if math.isnan(intermediate_value):
+                    score = (-step, float('inf'))
```

Comment:
I think `float('inf')` should be multiplied by `sign` to support maximization.

This is just my personal preference, but the following style may help developers understand the intention (i.e., NaN is regarded as the worst value).

```python
if math.isnan(intermediate_value):
    intermediate_value = float('inf')
score = (-step, sign * intermediate_value)
```

Reply:
I think the current code is correct.

If the `float('inf')` is multiplied by `sign` in maximization mode, the score becomes `(-step, -float('inf')`. 
It means that the trial (it was pruned after reporting a NaN value) has the best score with in the same step (this is unintended behavior).

Reply:
I see. This `score` is handled as minimization regardless of whether the study direction is `minimize` or `maximize`, and the current implementation is correct.
Thank you for your explanation.